### PR TITLE
Fix crate-name option in rustdoc

### LINKED ISSUE #51229: rustdoc errors on files with punctuation
`rustdoc main.rs --crate-name foo` succeeds and outputs documentation for a crate called `foo`, but `rustdoc main.bane.rs --crate-name foo` errors with "invalid character `.` in crate name: `main.bane`".

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #51229.

r? @QuietMisdreavus

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
Slightly alarming that the function at fault is the same one that rustc uses. I wonder if that function should be fixed as well?

To be sure, this PR still needs to land. It's weird that we're not handing the user-provided crate name to the compiler.

**Comment by GuillaumeGomez** [comment]
That's the case of most rustdoc options. The rustc function isn't at fault, the option handling on rustdoc's side is.

File: src/librustc/middle/cstore.rs [hunk 1]
```diff
@@ -276,7 +276,7 @@ pub fn validate_crate_name(sess: Option<&Session>, s: &str, sp: Option<Span>) {
         }
         for c in s.chars() {
             if c.is_alphanumeric() { continue }
-            if c == '_'  { continue }
+            if c == '_' { continue }
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustc/middle/cstore.rs]: Oups. Just went through the code trying to understand what was going on...

**Comment by QuietMisdreavus** [comment]
Thanks! r=me pending travis.

**Pull request merged**